core/src/main/java/org/apache/calcite/interpreter/Bindables.java

```diff
@@ -517,29 +521,76 @@ public Node implement(InterpreterImplementor implementor) {
     }
   }
 
-  /**
-   * Rule to convert an {@link org.apache.calcite.rel.logical.LogicalUnion}
-   * to a {@link BindableUnion}.
-   */
-  public static class BindableUnionRule extends ConverterRule {
+  @Deprecated // use BindableSetOpRule instead
+  public static class BindableUnionRule extends BindableSetOpRule {
 
     /**
      * Creates a BindableUnionRule.
      *
      * @param relBuilderFactory Builder for relational expressions
      */
     public BindableUnionRule(RelBuilderFactory relBuilderFactory) {
-      super(LogicalUnion.class, (Predicate<RelNode>) r -> true,
+      super(relBuilderFactory);
+    }
+  }
+
+  @Deprecated // use BindableSetOpRule instead
+  public static class BindableIntersectRule extends BindableSetOpRule {
+
+    /**
+     * Creates a BindableIntersectRule.
+     *
+     * @param relBuilderFactory Builder for relational expressions
+     */
+    public BindableIntersectRule(RelBuilderFactory relBuilderFactory) {
+      super(relBuilderFactory);
+    }
+  }
+
+  @Deprecated // use BindableSetOpRule instead
+  public static class BindableMinusRule extends BindableSetOpRule {
+
+    /**
+     * Creates a BindableMinusRule.
+     *
+     * @param relBuilderFactory Builder for relational expressions
+     */
+    public BindableMinusRule(RelBuilderFactory relBuilderFactory) {
+      super(relBuilderFactory);
+    }
+  }
```

Comment:
There is no BindableIntersectRule, BindableMinusRule before, why do you need this?

Reply:
Thanks for review, they should not exist, and have updated it.